File: TheSwitcher_ObjC/TheSwitcher_ObjC/ViewControllers/DetailViewController.m
//
//  DetailViewController.m
//  TheSwitcher_ObjC
//
//  Created by Fábio Maciel de Sousa on 10/02/21.
//

#import "DetailViewController.h"

@interface DetailViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *lampImageView;
@property (weak, nonatomic) IBOutlet UILabel *divisionTextLabel;
@property (weak, nonatomic) IBOutlet UILabel *divisionStatusLabel;
@end

@implementation DetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.lampImageView.image = [UIImage imageNamed: self.status ? @"light_image_ON" : @"light_image_OFF"];
    self.divisionTextLabel.text = [NSString stringWithFormat: @"Your %@ light is", self.division];
    self.divisionStatusLabel.text = self.status ? @"ON" : @"OFF";
    self.navigationItem.title = self.division;
}


@end

File: TheSwitcher_FabioMs/TheSwitcher_FabioMs/DetailViewController.swift
//
//  DetailViewController.swift
//  TheSwitcher_FabioMs
//
//  Created by Fábio Maciel de Sousa on 10/02/21.
//

import UIKit

class DetailViewController: UIViewController {
    @IBOutlet weak private var lampImageView: UIImageView!
    @IBOutlet weak private var divisionTextLabel: UILabel!
    @IBOutlet weak private var lampStateLabel: UILabel!
    
    var division = ""
    var lampState = false
    
    override func viewDidLoad() {
        super.viewDidLoad()
        divisionTextLabel.text = division.formattedDivision
        lampStateLabel.text = lampState ? "ON" : "OFF"
        lampImageView.image = lampState ? #imageLiteral(resourceName: "light_image_ON") : #imageLiteral(resourceName: "light_image_OFF")
        navigationItem.title = division
    }

}
